### PR TITLE
Support serialising custom types

### LINKED ISSUE #30: Feature request: custom field values using valuable crate
Hi! I'm currently using this crate for my JSON output but want to try adding custom structured values using tracing's unstable `valuable` feature flag ([in their docs](https://docs.rs/tracing/0.1.37/tracing/index.html#unstable-features)).

Looks like `tracing-bunyan-formatter` doesn't support that currently but it could be added fairly easily I think under a feature flag. I would probably use this crate to serialize the `Valuable` structs: <https://crates.io/crates/valuable-serde>.

What do you think?. Would you accept a PR for this?

### PR DESCRIPTION
**Pull request by fluffysquirrels**
As discussed in #30, this adds support for serialising custom types that implement `valuable::Valuable` behind the feature flag `valuable`.

This PR adds:

* The feature
* Feature flag documentation in `README.md`
* Example in `examples/valuable.rs`
* Unit tests in `tests/e2e.rs`
* A new CI build that runs with the `valuable` feature enabled to run the new test and example. [All builds passing in CircleCI on my fork](https://app.circleci.com/pipelines/github/fluffysquirrels/tracing-bunyan-formatter/3/workflows/4c9fd1ba-b8d7-4ef7-982d-d57b4d0ef64b).
* Reworked tests slightly to no longer use a shared buffer. Tests can now run in parallel: this is faster, and you don't have to remember to add `--test-threads 1` to test commands.

Closes #30.

### CODE REVIEW

File: src/storage_layer.rs [hunk 1]
```diff
@@ -92,6 +92,16 @@ impl Visit for JsonStorage<'_> {
...
         };
     }
+
+    #[cfg(all(tracing_unstable, feature = "valuable"))]
+    #[cfg_attr(docsrs, doc(cfg(all(tracing_unstable, feature = "valuable"))))]
+    fn record_value(&mut self, field: &Field, value: valuable::Value<'_>) {
+        let serializable = valuable_serde::Serializable::new(value);
+        let json_value = serde_json::to_value(serializable).unwrap_or(serde_json::Value::String(
+            "serde_json Serialization error".to_string(),
+        ));
```
Comment (LukeMathWalker) [inline src/storage_layer.rs:102]: I'd prefer to skip the value if serialization fails, rather than use a sentinel value.

**Comment by LukeMathWalker** [comment]
It looks great!
Just one comment on serialization failures 👌🏻

File: src/storage_layer.rs (hunk 1 above)
Comment (fluffysquirrels) [reply src/storage_layer.rs:102]: Sure, makes sense. I'll make the change and push it later. Probably later today.

Comment (LukeMathWalker) [reply src/storage_layer.rs:102]: That would work!

Comment (fluffysquirrels) [reply src/storage_layer.rs]: Done in latest. [Link](https://github.com/fluffysquirrels/tracing-bunyan-formatter/blob/9e8abf540feed97e2da05b52cc096129a1b04ff2/src/storage_layer.rs#L101-L118)

**Comment by LukeMathWalker** [comment]
Looking good, thanks!

**Pull request merged**